lib/tap.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+var format = require('util').format;
+
+function getSourceFromStack(stack) {
+	return stack
+		.split('\n')
+		.slice(2, 3)
+		.join('')
+		.replace(/^\s+at /, '');
+}
+
+exports.start = function () {
+	console.log('TAP version 13');
+};
+
+var i = 0;
+
+exports.test = function (test) {
+	var output = [];
+
+	output.push('# ' + test.title);
+
+	var asserts = test.asserts;
+
+	// if test failed, pop last assertion, because the error is in `test.error`
+	if (test.error) {
+		asserts.pop();
+	}
+
+	asserts.forEach(function (assert) {
+		output.push(format('ok %d - %s', ++i, assert.message));
+	});
+
+	if (test.error) {
+		output.push(format('not ok %d - %s', ++i, test.error.message));
+		output.push('  ---');
+		output.push('    operator: ' + test.error.operator);
+		output.push('    expected: ' + test.error.expected);
+		output.push('    actual: ' + test.error.actual);
+		output.push('    at: ' + getSourceFromStack(test.error.stack));
+		output.push('  ...');
+	}
+
+	console.log(output.join('\n'));
```

Comment:
Need to collect output first and then display it all at once to avoid collisions with other concurrent `console.log()`s.